inference-engine/src/gna_plugin/optimizer/gna_pass_manager.cpp

```diff
@@ -230,6 +231,13 @@ static std::vector<CNNLayerPtr> getCandidatesForIdentityInsertion(const CNNLayer
                 prevLayers.push_back(CNNNetPrevLayer(l, i));
             }
         }
+    } else if (gemm != nullptr) {
+        for (int i = 0; CNNNetHasPrevLayer(l.get(), i); ++i) {
```

Comment:
why is Gemm case not covered by 'else' section? Maybe we can just modify line 248 to check correct input:
auto prevLayer = PrevFunctionalLayer(l, LayerInfo(l).isGemm() ? 1 : 0);
I don't think that we will need to insert Identity between the const and the gemm layers since const has 16-bit output.